rfd/0090-db-mfa-sessions.md

```diff
@@ -0,0 +1,176 @@
+---
+author: Gavin Frazar (gavin.frazar@goteleport.com)
+state: draft
+---
+ 
+# RFD 90 - Database MFA Sessions
+
+## Required Approvers
+- Engineering: Marek `@smallinsky`, Roman `@r0mant`
+- Product: Sasha `@klizhentas`, Xin `@xinding33`
+
+## What
+Prevent database access users from being required to tap an MFA device every minute when per-session-MFA is enabled.
+
+## Why
+Database access via GUI client such as DataGrip requires the user to start a local proxy to connect to: `tsh proxy db --tunnel <db>`.
+
+To support per-session-MFA for the local proxy, we are adding a callback function called on each new connection to the local proxy.
+This callback function will check if the database certs need to be reissued to proxy the connection, and prompt for credentials/MFA if needed.
+
+Currently, database certs are issued with 1 minute ttl if per-session-MFA is enabled as described in [RFD 14](https://github.com/gravitational/teleport/blob/master/rfd/0014-session-2FA.md).
+This causes poor UX when the GUI client can establish a new connection at any time,
+and a user may be prompt once per minute for their MFA.
+
+We should remove the 1 minute restriction on database cert TTL to improve UX.
+
+## Details
+When per-session-MFA is enabled, we should not restrict database cert TTL to 1 minute.
+
+Instead, database cert TTL should be restricted to `max_session_ttl`, and the cert
```

Comment:
I think that using current default behavior where cert expiration time is set based on `max_session_ttl` is `reasonable`. Alternatively the DB MFA certs expiration time can be unset allowing to run tsh proxy db forever without second MFA prompt though we will ignore basic teleport cluster restriction in this case.

Reply:
Yes, in Connect when the tsh daemon process becomes aware of this, it can notify the Electron app about it and then the app can just bring [its window to the top](https://www.electronjs.org/docs/latest/api/browser-window#winfocus). This is how we decided to solve it for now, not only for per session MFA but also for things like trying to connect to a proxy after your cert has expired.